ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/desc/formatter/TextDescTableFormatter.java

```diff
@@ -0,0 +1,575 @@
+    displayAllParameters(params, tableInfo, true, false);
+  }
+
+  private void displayAllParameters(Map<String, String> params, StringBuilder tableInfo, boolean escapeUnicode,
+      boolean isOutputPadded) {
+    List<String> keys = new ArrayList<String>(params.keySet());
+    Collections.sort(keys);
+    for (String key : keys) {
+      String value = params.get(key);
+      if (key.equals(StatsSetupConst.NUM_ERASURE_CODED_FILES)) {
+        if ("0".equals(value)) {
+          continue;
+        }
+      }
+      tableInfo.append(FIELD_DELIM); // Ensures all params are indented.
+      formatOutput(key, escapeUnicode ? StringEscapeUtils.escapeJava(value) : HiveStringUtils.escapeJava(value),
+          tableInfo, isOutputPadded);
+    }
+  }
+
+  private String getConstraintsInformation(Table table) {
+    StringBuilder constraintsInfo = new StringBuilder(DEFAULT_STRINGBUILDER_SIZE);
+
+    constraintsInfo.append(LINE_DELIM + "# Constraints" + LINE_DELIM);
+    if (PrimaryKeyInfo.isPrimaryKeyInfoNotEmpty(table.getPrimaryKeyInfo())) {
+      constraintsInfo.append(LINE_DELIM + "# Primary Key" + LINE_DELIM);
+      getPrimaryKeyInformation(constraintsInfo, table.getPrimaryKeyInfo());
+    }
+    if (ForeignKeyInfo.isForeignKeyInfoNotEmpty(table.getForeignKeyInfo())) {
+      constraintsInfo.append(LINE_DELIM + "# Foreign Keys" + LINE_DELIM);
+      getForeignKeysInformation(constraintsInfo, table.getForeignKeyInfo());
+    }
+    if (UniqueConstraint.isUniqueConstraintNotEmpty(table.getUniqueKeyInfo())) {
+      constraintsInfo.append(LINE_DELIM + "# Unique Constraints" + LINE_DELIM);
+      getUniqueConstraintsInformation(constraintsInfo, table.getUniqueKeyInfo());
+    }
+    if (NotNullConstraint.isNotNullConstraintNotEmpty(table.getNotNullConstraint())) {
+      constraintsInfo.append(LINE_DELIM + "# Not Null Constraints" + LINE_DELIM);
+      getNotNullConstraintsInformation(constraintsInfo, table.getNotNullConstraint());
+    }
+    if (DefaultConstraint.isCheckConstraintNotEmpty(table.getDefaultConstraint())) {
+      constraintsInfo.append(LINE_DELIM + "# Default Constraints" + LINE_DELIM);
+      getDefaultConstraintsInformation(constraintsInfo, table.getDefaultConstraint());
+    }
+    if (CheckConstraint.isCheckConstraintNotEmpty(table.getCheckConstraint())) {
+      constraintsInfo.append(LINE_DELIM + "# Check Constraints" + LINE_DELIM);
+      getCheckConstraintsInformation(constraintsInfo, table.getCheckConstraint());
+    }
+    return constraintsInfo.toString();
+  }
+
+  private void getPrimaryKeyInformation(StringBuilder constraintsInfo, PrimaryKeyInfo constraint) {
+    formatOutput("Table:", constraint.getDatabaseName() + "." + constraint.getTableName(), constraintsInfo);
+    formatOutput("Constraint Name:", constraint.getConstraintName(), constraintsInfo);
+    Map<Integer, String> columnNames = constraint.getColNames();
+    String title = "Column Name:".intern();
```

Comment:
`intern` of a constant values is probably useless.  Please remove.

Reply:
Removed.